src/QsCompiler/Tests.Compiler/LambdaLiftingTests.fs

```diff
@@ -0,0 +1,573 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Testing
+
+open System
+open System.IO
+open System.Text.RegularExpressions
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.CompilationBuilder
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations.LiftLambdas
+open Microsoft.Quantum.QsCompiler.Transformations.QsCodeOutput
+open Xunit
+open System.Collections.Immutable
+
+type ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>
+
+type LambdaLiftingTests() =
+
+    let compilationManager = new CompilationUnitManager(ProjectProperties.Empty, (fun ex -> failwith ex.Message))
+
+    let getTempFile () =
+        new Uri(Path.GetFullPath(Path.GetRandomFileName()))
+
+    let getManager uri content =
+        CompilationUnitManager.InitializeFileManager(
+            uri,
+            content,
+            compilationManager.PublishDiagnostics,
+            compilationManager.LogException
+        )
+
+    let ReadAndChunkSourceFile fileName =
```

Comment:
These should somehow be private static member functions, which should have PascalCase. How do I specify them as static private members in F#?

Reply:
> These should somehow be private static member functions

I don't think they should - they don't need to access any other instance member, so they work fine as `let`s.

Reply:
I guess they don't really work well as `let` bindings if you are asserting that they would then need to be camelCase, which isn't desirable, and they being let bindings doesn't accurately my intention for them. My suggestion for them to be static members is a compromise.

Reply:
I don't understand why `camelCase` is not desirable? That is the standard F# naming convention.

Reply:
`camelCase` is something that should be used to distinguish local variables. `PascalCase` is something that should be used to distinguish functions. It is very useful to be able to distinguish between things that are scoped to the immediate function versus things that are available to a wider scope.

Reply:
Note that that's a C# naming convention - since this is F# code, it's preferrable to follow the conventions of the language being used. In this case, the `let`s are local variables (they are scoped to the constructor - in F# all members are also in the constructor's scope). IMO it's somewhat less idiomatic to make them private static members but that would also work.

> How do I specify them as static private members in F#?

Here's the documentation on defining methods and access control:

* https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/methods
* https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/access-control

Reply:
I had hoped this would become a moot discussion once I moved these items to the TestUtils, but the flaw in this F# standard seems to be even more egregious now. These functions are now defined in a totally different module as publicly available members of that module, and act as public static members (there should be some way of specifying that they're static). They are even more removed from the places in which they are called now, and are not in any way local. This should be made clear with their name.
I should also qualify them with the module name. Let me make that change quick.

Reply:
Hmm, I'm not sure I'm following exactly... even in C#, camelCase and PascalCase doesn't distinguish local/non-local bindings. For example, private fields in C# classes are `camelCase` even though they are class-scoped. Also, local functions in C# are `PascalCase`, even though they are scoped to a single method.

In any case it's generally a good idea to follow the conventions of the language being used, even if it's different from what you're used to in other languages.